File: test/Integration/test-csharp/AddProductColumnTypes.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Common;

namespace Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Integration
{
    public static class AddProductColumnTypes
    {
        /// <summary>
        /// This function is used to test compatability with converting various data types to their respective
        /// SQL server types.
        /// </summary>
        [FunctionName(nameof(AddProductColumnTypes))]
        public static IActionResult Run(
                [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "addproduct-columntypes")] HttpRequest req,
                [Sql("dbo.ProductsColumnTypes", "SqlConnectionString")] out ProductColumnTypes product)
        {
            product = new ProductColumnTypes()
            {
                ProductId = int.Parse(req.Query["productId"]),
                BigInt = int.MaxValue,
                Bit = true,
                DecimalType = 1.2345M,
                Money = 1.23M,
                Numeric = 1.2345M,
                SmallInt = 0,
                SmallMoney = 1.23M,
                TinyInt = 1,
                FloatType = 1.2,
                Real = 1.2f,
                Date = DateTime.Now,
                Datetime = DateTime.Now,
                Datetime2 = DateTime.Now,
                DatetimeOffset = DateTime.Now,
                SmallDatetime = DateTime.Now,
                Time = DateTime.Now.TimeOfDay,
                CharType = "test",
                Varchar = "test",
                Nchar = "\u2649",
                Nvarchar = "\u2649",
            };

            // Items were inserted successfully so return success, an exception would be thrown if there
            // was any issues
            return new OkObjectResult("Success!");
        }
    }
}

File: test/Integration/test-java/src/main/java/com/function/AddProductColumnTypes.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.function;

import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.OutputBinding;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.sql.annotation.SQLOutput;
import com.function.Common.ProductColumnTypes;

import java.math.BigDecimal;
import java.sql.Timestamp;
import java.sql.Time;
import java.util.Optional;

public class AddProductColumnTypes {
    @FunctionName("AddProductColumnTypes")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "addproduct-columntypes")
                HttpRequestMessage<Optional<String>> request,
            @SQLOutput(
                name = "product",
                commandText = "dbo.ProductsColumnTypes",
                connectionStringSetting = "SqlConnectionString")
                OutputBinding<ProductColumnTypes> product) {

        ProductColumnTypes p = new ProductColumnTypes(
            Integer.parseInt(request.getQueryParameters().get("productId")),
            (long)999,
            true,
            new BigDecimal("1.2345"),
            new BigDecimal("1.2345"),
            new BigDecimal("1.2345"),
            (short)1,
            new BigDecimal("1.2345"),
            (short)1,
            0.1,
            0.1,
            new Timestamp(System.currentTimeMillis()).toString(),
            new Timestamp(System.currentTimeMillis()).toString(),
            new Timestamp(System.currentTimeMillis()).toString(),
            new Timestamp(System.currentTimeMillis()).toString(),
            new Timestamp(System.currentTimeMillis()).toString(),
            new Time(System.currentTimeMillis()).toString(),
            "test",
            "test",
            "\u2649",
            "\u2649",
            "dGVzdA==",
            "dGVzdA==");
        product.setValue(p);

        // Items were inserted successfully so return success, an exception would be thrown if there
        // was any issues
        return request.createResponseBuilder(HttpStatus.OK).header("Content-Type", "application/json").body(product).build();
    }
}
